kolibri_server_setup.py

```diff
@@ -75,13 +76,37 @@ def enable_cherrypy():
     update_options_file('Server', "CHERRYPY_START", True, KOLIBRI_HOME)
 
 
+def delete_redis_cache():
+    """
+    Delete previous cache in redis to reset it when the service starts.
+    The purpose is avoiding redis memory usage growing infinitely.
+    """
+    redis_args = [
+        (str(redis_db), ":1:views.decorators.*"),
+        (str(redis_db), ":1:*_dataset"),
+        (str(redis_db), ":1:content_cache_key"),
+        (str(redis_db), ":1:device_settings_cache_key"),
+        (str(redis_db + 1), "built_files:1:*"),
+    ]
+    try:
+        for arg in redis_args:
+            search = ["redis-cli", "-n", arg[0], "--scan", "--pattern", arg[1]]
+            delete = ["xargs", "redis-cli", "-n", arg[0], "unlink"]
+            exe_search = subprocess.Popen(search, stdout=subprocess.PIPE)
+            subprocess.Popen(delete, stdin=exe_search.stdout)
+    except:
```

Comment:
You have a bare exception clause here :)

Reply:
Yes, I know. I am tempted of removing the `try..except` clause there. I was trying to make it fail and thinking of possible exceptions to catch but I could not find any? any ideas or do you think removing the `try` is fine?

Reply:
What if `redis-cli` isn't in the path? What if in 3 months some update causes a `KeyError`?

If you know exactly what exit codes you want to expect, then those can be expected and the rest would be a crash?

If this is supposed to give a guarantee that the redis queue is purged periodically and doesn't grow infinitely, then the guarantee isn't 100% unless it's allowed to fail loudly :)

Reply:
> What if `redis-cli` isn't in the path? What if in 3 months some update causes a `KeyError`?
> 

`redis-cli` is in `redis-tool `which is a dependency of `redis-server` which is a dependency of kolibri-server, so if kolibri-server is installed, it's in the path.

> If you know exactly what exit codes you want to expect, then those can be expected and the rest would be a crash?
> 

using subprocess the code is not catching error codes, it's not a task made in python, but in the OS


> If this is supposed to give a guarantee that the redis queue is purged periodically and doesn't grow infinitely, then the guarantee isn't 100% unless it's allowed to fail loudly :)

no, it does not purge it periodically but it warranties it will purge it if you restart the service or the server what is the first thing users do when they have problems.

Reply:
> no, it does not purge it periodically but it warranties it will purge it if you restart the service or the server what is the first thing users do when they have problems.

That's what I meant, I just mean that when exceptions are silenced, then it's hard to guarantee anything, since we won't know if it fails.

Reply:
> That's what I meant in regards to "periodically".

Added a weekly cron to delete the redis cache

Reply:
Awesome!